From 85420d5d4f78a6ba4cf70ed8ee092e0ef4fb9a54 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@kde.org>
Date: Mon, 24 Aug 2026 16:21:29 +0200
Subject: [PATCH] QIconLoader: Don't consider fallbackThemeName() in
 themeName()

themeName and fallbackThemeName are independent things.

bebbf84167a4341eed27ecfca480ff62dfa1b159 mixes the two together,
which results in setFallbackThemeName() incorrectly changing
themeName().

Restore that part to how it was before.

Amends bebbf84167a4341eed27ecfca480ff62dfa1b159

Fixes: QTBUG-149431
Pick-to: 6.12 6.11
Change-Id: I7bb5e4329b677abb95b74269877f8d9c0222a972
Reviewed-by: Axel Spoerl <axel.spoerl@qt.io>
---
 src/gui/image/qiconloader.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/gui/image/qiconloader.cpp b/src/gui/image/qiconloader.cpp
index 62353ec9cd9..4526c6f75fb 100644
--- a/src/gui/image/qiconloader.cpp
+++ b/src/gui/image/qiconloader.cpp
@@ -151,8 +151,7 @@ QString QIconLoader::themeName() const
         return m_userTheme;
 
     if (m_systemTheme.isEmpty()) {
-        const QString &themeName = systemThemeName();
-        m_systemTheme = !themeName.isEmpty() ? themeName : fallbackThemeName();
+        m_systemTheme = systemThemeName();
     }
     return m_systemTheme;
 }
-- 
GitLab

